web-marketplace/src/legacy-pages/Dashboard/Dashboard.tsx

```diff
@@ -155,37 +354,76 @@ export const Dashboard = () => {
     window.scrollTo({ top: 0, behavior: 'instant' });
   }, [pathname]);
 
-  const resolvedAddress = useMemo(() => {
-    const walletAddress = getWalletAddress({ activeAccount, wallet });
-    if (walletAddress) return walletAddress;
-
-    return privActiveAccount?.email || '';
-  }, [activeAccount, wallet, privActiveAccount?.email]);
+  const viewProfileAccount = useMemo<Pick<
+    Account,
+    'addr' | 'id' | 'name' | 'type' | 'image'
+  > | null>(() => {
+    if (selectedAccount?.type === 'org') {
+      return {
+        id: selectedAccount.roleAccountId ?? selectedAccount.id,
```

Comment:
`selectedAccount.roleAccountId` is the current user account id so this is buggy since this would lead to the user profile page instead of the org profile page
I don't think we need this `roleAccountId` field at all